docs(NODE-5080): add a note about class syntax (#565)
mongodb/js-bson: docs/upgrade-to-v5.md

> This will preserve newer ECMAScript 2020 features like optional chaining, nullish coalescing, export \* as ns, and dynamic import(...) syntax. It also means bigint literals now have a stable target below esnext.

#### `class` syntax

> Previously, our compilation target was set such that our BSON `class`(s) became functions in the shipped code, allowing callers to omit the `new` keyword. As a result of the above `"target"` upgrade the library now ships javascript `class` syntax which requires callers to use the `new` keyword.

Reviewer: ```suggestion
Previously, our compilation target was set such that each BSON `class` became a function in the shipped code, allowing callers to omit the `new` keyword. Because of the `"target"` upgrade, the library now ships javascript `class` syntax which requires callers to use the `new` keyword.
```
